crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -500,6 +500,22 @@ impl UncompressedDataManager {
         // If not able to find any in-memory finished segments, panic!() is the only option.
         panic!("Not enough reserved memory for the necessary uncompressed buffers.");
     }
+
+    /// Change the amount of memory for uncompressed data in bytes according to `value_change`. If
+    /// less than zero bytes remain, spill uncompressed data to disk. If all the data is spilled
+    /// successfully return [`Ok`], otherwise return [`IOError`].
+    pub(super) async fn set_uncompressed_remaining_memory_in_bytes(&mut self, value_change: isize) {
+        let mut current_remaining_memory: isize =
+            self.uncompressed_remaining_memory_in_bytes as isize + value_change;
+
+        while current_remaining_memory < 0 {
```

Comment:
do we risk an infinite loop here if the `value_change` is a large negative number? Is it guaranteed that `spill_finished_buffer` can free enough?

Reply:
Usually since `value_change` is based on the new value given to the configuration manager, the negative value can not be large enough to cause any issues. However, if a large negative number is given directly to this function, it would spill data until there is no longer any data to spill, and then it would panic and stop execution of the program. This is the intended way to handle this and we test this in the `test_panic_if_decreasing_uncompressed_remaining_memory_in_bytes_below_zero` unit test.

Reply:
OK, that is good then